database/factories/AppointmentFactory.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Lightit\Appointments\Domain\Models\Appointment;
+use Lightit\Clinics\Domain\Models\Clinic;
+use Lightit\Doctors\Domain\Models\Doctor;
+use Lightit\Users\Domain\Models\User;
+
+/**
+ * @extends Factory<Appointment>
+ */
+class AppointmentFactory extends Factory
+{
+    protected $model = Appointment::class;
+
+    public function definition(): array
+    {
+        return [
+            'start_time' => $start = fake()->dateTime(),
+            'end_time' => (clone $start)->modify('+' . fake()->numberBetween(15, 100) . 'minutes'),
```

Comment:
This is much cleaner in my opinion because of the following reasons:
- For dates, we use CarbonImmutable (you don't need to use `clone` with it
- In my opinion, it's better to set a fixed start time since it's more controlled
- It's better to use variables instead of all the operations in the line because, through the variable name, other developers (and you in the future) can understand what you're doing there

```suggestion
    {
        $appointmentDurationInMinutes = rand(15, 100);
        $startTime = CarbonImmutable::now();
        $endTime = $startTime->addMinutes($appointmentDurationInMinutes);
    
        return [
            'start_time' => $startTime,
            'end_time' => $endTime,
```

Reply:
Your right, this is much cleaner, thanks!